includes/class-wc-payments-payment-method-messaging-element.php

```diff
@@ -118,26 +123,33 @@ public function init() {
 			WC_Payments::get_file_version( 'dist/product-details.css' ),
 		);
 
+		$country = empty( $billing_country ) ? $store_country : $billing_country;
+
+		$script_data = [
+			'productId'         => 'base_product',
+			'productVariations' => $product_variations,
+			'country'           => $country,
+			'locale'            => WC_Payments_Utils::convert_to_stripe_locale( get_locale() ),
+			'accountId'         => $this->account->get_stripe_account_id(),
+			'publishableKey'    => $this->account->get_publishable_key( WC_Payments::mode()->is_test() ),
+			'paymentMethods'    => array_values( $bnpl_payment_methods ),
+			'currencyCode'      => $currency_code,
+			'isCart'            => is_cart(),
+			'isCartBlock'       => $is_cart_block,
+			'cartTotal'         => WC_Payments_Utils::prepare_amount( $cart_total, $currency_code ),
+			'nonce'             => wp_create_nonce( 'wcpay-bnpl-nonce' ),
```

Comment:
I noted this in class-wc-payments-utils.php as well, but I think we should separate the nonces here.